core/sql/generator/GenRelScan.cpp

```diff
@@ -1182,6 +1181,15 @@ if (hTabStats->isOrcFile())
 
   UInt32 rangeTailIOSize = (UInt32)
       CmpCommon::getDefaultNumeric(HDFS_IO_RANGE_TAIL);
+  if (rangeTailIOSize == 0) 
+    {
+      rangeTailIOSize = getTableDesc()->getNATable()->getRecordLength() +
```

Comment:
This record length is what the compiler thinks, the real length may be more or less. Hive makes it probably more complex, some fields like date/time/timestamp are strings in Hive and encoded binaries in Trafodion. Character sets may be different as well in some cases (e.g. GBK). I don't have a better solution than what the code does, but let's say we have this: create table t(a char(20000) not null, b timestamp(6) not null, c timestamp(6) not null). Trafodion will make the length 20022. In reality, a string may look like "20k chars..., 2015-01-01 00:00:00.000, 2016-01-01 00:00:00.000". That's 2049 characters. Maybe the solution is to add the 16K to the record length, not take the max?